fontbe/src/metrics_and_limits.rs

```diff
@@ -0,0 +1,186 @@
+//! Generates the [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx) and
+//! [hhea](https://learn.microsoft.com/en-us/typography/opentype/spec/hhea) tables
+
+use std::cmp::{max, min};
+
+use fontdrasil::orchestration::Work;
+use read_fonts::types::FWord;
+use write_fonts::{
+    dump_table,
+    tables::{glyf::Contour, hhea::Hhea, hmtx::Hmtx, vmtx::LongMetric},
+    OtRound,
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, Glyph},
+};
+
+struct MetricAndLimitWork {}
+
+pub fn create_metric_and_limit_work() -> Box<BeWork> {
+    Box::new(MetricAndLimitWork {})
+}
+
+#[derive(Debug, Default)]
+struct GlyphLimits {
+    min_left_side_bearing: Option<i16>,
+    min_right_side_bearing: Option<u16>,
+    x_max_extent: Option<i16>,
+    advance_width_max: u16,
+    max_points: u16,
+    max_contours: u16,
+    max_component_elements: u16,
+}
+
+impl GlyphLimits {
+    fn update(&mut self, advance: u16, glyph: &Glyph) {
+        // min side bearings are only for non-empty glyphs
+        // we will presume only simple glyphs with no contours are empty
+        if glyph.is_empty() {
+            return;
+        }
+
+        let bbox = glyph.bbox();
```

Comment:
since you are getting the glyph bboxes to do these limit computations, you may as well also collect maximum bounds (union of all non-empty glyph's bboxes) and use that to set the [head](https://learn.microsoft.com/en-us/typography/opentype/spec/head) table font-wide bounds (head.xMin, yMin, xMax, yMax).